src/Corax/IndexWriter.cs

```diff
@@ -299,6 +305,57 @@ public long Index(string id, Span<byte> data)
             return Index(idSlice, data);
         }
 
+        public unsafe long Index(string id, Span<byte> data, float documentBoost)
+        {
+            var entryId = Index(id, data);
+            AppendDocumentBoost(entryId, documentBoost);
+            return entryId;
+        }
+
+        //Document Boost should add priority to some documents but also should not be the main component of boosting.
+        //The natural logarithm slows down our scoring increase for a document so that the ranking calculated at query time is not forgotten.
+        private unsafe void AppendDocumentBoost(long entryId, float documentBoost, bool isUpdate = false)
+        {
+            if (documentBoost.AlmostEquals(1f))
+            {
+                
+                // We don't store `1` but if user update boost value to 1 we've to delete the previous one
+                if (isUpdate)
+                {
+                    _documentBoost.Read(entryId, out var previous);
+                    if (previous.HasValue)
+                        _documentBoost.Delete(entryId);
```

Comment:
```suggestion
                    _documentBoost.Delete(entryId);
```

Don't bother to read, just try to delete immediately.